File: app/src/main/java/com/stupidtree/hichat/ui/base/StringTrigger.java
package com.stupidtree.hichat.ui.base;

public class StringTrigger extends Trigger {
    String data;
    public static StringTrigger getActioning(String data){
        StringTrigger stringTrigger = new StringTrigger();
        stringTrigger.setActioning();
        stringTrigger.data = data;
        return stringTrigger;
    }

    public String getData() {
        return data;
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/base/StringTrigger.kt
package com.stupidtree.cloudliter.ui.base

class StringTrigger : Trigger() {
    var data: String? = null

    companion object {
        fun getActioning(data: String?): StringTrigger {
            val stringTrigger = StringTrigger()
            stringTrigger.setActioning()
            stringTrigger.data = data
            return stringTrigger
        }
    }
}